src/main/java/it/eng/idsa/dataapp/ftp/client/FTPClient.java

```diff
@@ -0,0 +1,113 @@
+package it.eng.idsa.dataapp.ftp.client;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.security.KeyPair;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.PreDestroy;
+
+import org.apache.sshd.client.SshClient;
+import org.apache.sshd.client.keyverifier.RequiredServerKeyVerifier;
+import org.apache.sshd.client.session.ClientSession;
+import org.apache.sshd.common.keyprovider.KeyIdentityProvider;
+import org.apache.sshd.scp.client.DefaultScpClientCreator;
+import org.apache.sshd.scp.client.ScpClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+import it.eng.idsa.dataapp.configuration.KeyStoreConfiguration;
+
+@Service
+public class FTPClient {
+
+	private static final Logger logger = LoggerFactory.getLogger(FTPClient.class);
+
+	@Value("${server.ssl.key-password}")
+	private String tlsKeystorePassword;
+
+	@Value("${application.dataLakeDirectory}")
+	String dataLakeDirectory;
+
+	@Value("${application.sftp.connectorId}")
+	private String username;
+
+	@Value("${application.sftp.host}")
+	private String host;
+
+	@Value("${application.sftp.port}")
+	private int port;
+
+	@Value("${application.sftp.defaultTimeoutSeconds}")
+	private long defaultTimeoutSeconds;
+
+	@Autowired
+	private KeyStoreConfiguration keyStoreConfiguration;
+
+	SshClient client;
```

Comment:
Not most elegant solution to use class member to store state. Not working well in multi threaded environment

Reply:
Thanks for the advice!

It will be refactored!